src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,32 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    /**
+     * $fieldPath List of paths on this object.
+     * @return mixed Returns DBField of the field on the object or null if there is not relations between fields.
+     */
+    private function hasRelations(string $fieldPath)
+    {
+        $component = $this;
+        foreach (explode('.', $fieldPath ?? '') as $relation) {
+            if (!$component) {
+                return false;
+            }
+
+            if (ClassInfo::hasMethod($component, $relation)) {
+                if ($component->hasField($relation)
+                    || (($component instanceof DataObject) && $component->getRelationType($relation) !== null)) {
+                    $component = $component->$relation();
+                } else {
+                    return false;
+                }
+            } else {
+                return true;
```

Comment:
The only purpose of this method is to additionally check that whatever `hasMethod` passes is really a relationship getter and not just a helper method. Therefore, assuming that all other checks are passed in `relObject` method and if they are passed, then `hasRelations ` returns `true` for them. If not passed, then this method will not be called.